pkg/util/iptables/iptables.go

```diff
@@ -129,6 +129,39 @@ const WaitSecondsMinVersion = "1.4.22"
 const WaitString = "-w"
 const WaitSecondsValue = "5"
 
+// MinMasqueradeFullyVersion is the minimum iptables version that supports the `--random-fully` modifier on `-j MASQUERADE`
+const MinMasqueradeRandomFullyVersion = "1.6.2"
+
+type TestOutcome struct {
+	Answer bool
+	Err    error
```

Comment:
er... no. Just use a multiple-value return from `VersionStringIsAtLeast`

Reply:
What is bad about defining a return type?  It has the advantage that a caller that wants just one member can make the call and select the member all in one expression; a multiple-value return forces the caller to write something more long-winded.

Reply:
At https://github.com/kubernetes/kubernetes/blob/655de9c860f1658ec86ded978dc99de8038e5c17/pkg/proxy/iptables/proxier_test.go#L376 you can see an example of the advantage of returning one struct rather than multiple values.

Reply:
It's non-go-like. But anyway, this is why I was saying we should do all the version checks once at the beginning and then just have "SupportsCheck()", "SupportsRandomFully()" etc methods. Then you don't need to worry about returning errors, because that will already have been dealt with a long time before.

#80368 refactors the utiliptables version checking and gets rid of `GetVersion()`. Originally it added a `SupportsCheck()` to replace it before we realized we didn't even need that. But you could set a `runner.supportsRandomFully` variable when first creating the object, and have a `SupportsRandomFully()` method that just returns that. (This also keeps all of the knowledge of specific version numbers inside utiliptables rather than having its consumers need to know them.)

Reply:
I see.  It looks like #80368 is establishing some groundwork to build on.